packages/connections/src/XHRConnection.js

```diff
@@ -0,0 +1,227 @@
+   * @param {string} [options.method=GET] – used method
+   * @param {*} [options.body] – payload for request
+   * @param {object} [options.headers] – additional headers (like content-type)
+   * @param {string} [options.responseType]
+   * @param {int} [timeout]
+   */
+  constructor(url, options = {}) {
+    super(url);
+
+    const {
+      method = "GET",
+      body = null,
+      headers = {},
+      responseType = "json",
+      timeout = 0
+    } = options;
+
+    if (!ALLOWED_METHODS.includes(method)) {
+      throw new Error(
+        `Invalid method "${method}". Valid methods are: "${ALLOWED_METHODS.join('", "')}".`
+      );
+    }
+
+    if (!ALLOWED_RESPONSE_TYPES.includes(responseType)) {
+      throw new Error(
+        `Invalid response type "${responseType}". Valid reponse types are "${ALLOWED_RESPONSE_TYPES.join('", "')}".`
+      );
+    }
+
+    if (typeof headers !== "object") {
+      throw new Error(
+        `Invalid headers. Type must be "object", is "${typeof headers}".`
+      );
+    }
+
+    if (!Number.isSafeInteger(timeout) || timeout < 0) {
+      throw new Error(
+        `Invalid timeout. Must be an integer >= 0 and less than Number.MAX_SAFE_INTEGER, is "${timeout}".`
+      );
+    }
+
+    /**
+     * @property {string}
+     * @protected
+     * @name XHRConnection#method
+     */
+    Object.defineProperty(this, "method", { value: method });
+
+    /**
+     * @property {*}
+     * @protected
+     * @name XHRConnection#body
+     */
+    Object.defineProperty(this, "body", { value: body });
+
+    /**
+     * @property {object}
+     * @protected
+     * @name XHRConnection#headers
+     */
+    Object.defineProperty(this, "headers", { value: headers });
+
+    /**
+     * @property {string}
+     * @protected
+     * @name XHRConnection#responseType
+     */
+    Object.defineProperty(this, "responseType", { value: responseType });
+
+    /**
+     * @property {int}
+     * @protected
+     * @name XHRConnection#timeout
+     */
+    Object.defineProperty(this, "timeout", { value: timeout });
+
+    /**
+     * @property {XMLHttpRequest}
+     * @protected
+     * @name XHRConnection#xhr
+     */
+    Object.defineProperty(this, "xhr", { value: new XMLHttpRequest() });
+  }
+
+  get response() {
+    if (this.xhr.response && this.xhr.responseType === this.responseType) {
+      return this.xhr.response;
+    }
+
+    /**
+     * IE11 doesnt support responseType="json" correctly,
+     * it will fall back to plain text, so we have to parse it.
+     */
+    if (
+      this.responseType === "json" &&
+      this.xhr.response &&
+      this.xhr.responseType === ""
+    ) {
+      return JSON.parse(this.xhr.response);
+    }
+
+    /**
+     * make sure there is no other weird browser behaviour
+     */
+    if (process.env.NODE_ENV !== "production") {
+      throw new TypeError(
+        `Unexpected XHR responseType (expected "${this.responseType}", was "${this.xhr.responseType}").`
```

Comment:
It throws in IE11 in dev mode because for whatever reason as soon as I add listener it calls `response` method and since the connection hasn't been open yet `this.xhr.response` is empty thus `IE11` specific code doesn't run, which results in the exception. Even though we're not developing in IE11 we still test in this browser time to time so it shouldn't happen.

Reply:
oh, I know why this happens! Looks like IE11 invokes the getter trying to dump the object in the console. But the thing is if you create an object and then try to use getter it will throw in _any_ browser, but actually it should just return `null`

Reply:
fixed 👍